2016/proposals/singh-shubham.md

```diff
@@ -0,0 +1,210 @@
+4. *Trainee Home page* -
+After getting accepted for the training program, the trainees recieve a randomly generated password, which they can use to log in to AMY, a trainee's home page is different from the AMY DASHBOARD, the AMY DASHBOARD is reservered for administrative purposes only. After logging in the trainees are directed to TRAINEE_HOME. This view will have a nav-bar - **HOME** , **ABOUT** and **NOTIFICATION**.
+  * *HOME*  shows the trainees progress, through a progress bar indicating which stages have been completed and the ones left. In this view the trainee can also request a discussion session by filling in a webform, with fields such as TOPIC, PREFERRED TIME, and so on.
+This view will also list down the general guidelines for instructor training, informing what is required in each stage of the training.
+  * *ABOUT* , this view shows the information entered by the trainee in their initial application, they can also edit this information, and change passwords here.
+  * *NOTIFICATION*,  this view allows the trainee to view all the comments/feedback from the discussion leaders and the lesson maintainers.In the format *comment: by whom: timestamp*.
+
+
+## Technical Details
+The Software stack for this project can be divided into:
+
+ * Backend  : [Django](https://www.djangoproject.com/)
+ * Database : [SQLite](https://www.sqlite.org/)
+ * Frontend : [Bootstrap](http://getbootstrap.com/),[JQuery](https://jquery.com/) for AJAX requests
+
+### Features and Model Schemes
+
+Trainee would be the main Model, maintaining most of the Flags for Progress and other information:
+
+```python
+lass Trainee(AbstractBaseUser):
+    '''Model for a Trainee '''
+    # listing just essential fields, more fields can and should be added when implementing
+    COURSE_CHOICES = (
+        ('1', 'SWC'),
+        ('2', 'DC'),
+        )
+    personal    = models.CharField(max_length=STR_LONG)
+    middle      = models.CharField(max_length=STR_LONG, null=True, blank=True)
+    family      = models.CharField(max_length=STR_LONG)
+    email       = models.CharField(max_length=STR_LONG, unique=True, null=True, blank=True)
+    course      = models.CharField(max_length=STR_LONG, choices=COURSE_CHOICES, null=False, default='1')
+    group       = models.ForeignKey(Trainee_Group,default=None,null=True)
+    Attendance_Flag = models.BooleanField(default=False) # maintains attendance record
+    Discussion_Flag = models.BooleanField(default=False) # maintains PASS or FAIL
+    HomeWork_Flag = models.BooleanField(default=False) # maintains PASS or FAIL
+
+    username    = models.CharField(
+        max_length=STR_MED, unique=True,
+        validators=[RegexValidator(r'^[\w\-_]+$', flags=re.A)],
+    )
+ ```
+
+Maintaining Group information:
+
+```python
+class Trainee_Group(models.Model):
+
+    instructor_assigned=models.ForeignKey(Person)
+    primary_contact_personal    = models.CharField(max_length=STR_LONG)
```

Comment:
The primary contact should be a member of the group.
While filling the form, they could nominate one of themselves to be the primary contact. Since this happens while filling the form this constraint is automatically enforced.